### ISSUE TITLE
sudo initialization

### ISSUE THREAD

**Issue by figue** [rmarquis/pacaur#418]
sudo initialization force to authenticate always (it executes "sudo -v").

We can discuss if it's a necessary "security" check, but pacaur only needs sudo to launch pacman. So if a user has pacman in sudoers with NOPASSWD, pacaur will ask for sudo authentication without a reason.

A simple patch could be done instead:
```
 788     # initialize sudo
 789     if sudo $pacmanbin -V > /dev/null; then
 790         [[ $sudoloop = true ]] && SudoV &
 791     fi
```

It remains SudoV function, which could be patched as well (it doesn't have any sense to activate that option).

PS: Another way to check if pacman is in sudoers could be:
`sudo -l | grep -q 'NOPASSWD: /usr/bin/pacman`

**Comment by rmarquis** [comment]
Good catch. I tested with `%wheel ALL=(ALL) NOPASSWD: ALL` but never realized the password was required even if pacman was set to `NOPASSWD`.

However, I do have the following in my output with your change:
```
sudo: a password is required
```
although it doesn't seem to have any negative effect.

The other proposed solution doesn't take into account more detailed pacman settings (f.e.  `NOPASSWD: /usr/bin/pacman -Syu` vs `NOPASSWD: /usr/bin/pacman -D`, or any other combination), so it's probably not robust enough.

> It remains SudoV function, which could be patched as well (it doesn't have any sense to activate that option).

Care to elaborate here? I'm not sure what you mean here.

**Comment by figue** [comment]
> sudo: a password is required
I've noticed thet if in my sudoers I have: "figue ALL=(ALL) NOPASSWD:/usr/bin/pacman" no password is required. But with groups doesn't work, for instance with "%wheel ALL=(ALL) NOPASSWD: ALL". But it works if you disable asking passwords for your user: "Defaults:figue      !authenticate"

The other proposed solution could be not robust... But even if you add settings after /usr/bin/pacman still could work:
```
$ sudo -l | grep 'NOPASSWD: /usr/bin/pacman'
    (ALL) NOPASSWD: /usr/bin/pacman -D
```
but I agree that executing a simple pacman -V is easier.

SudoV function: I wanted to say if you have sudoloop=true you still will be prompted for the password, because SudoV uses "sudo -n -v" to avoid the timeout. So we could add the same "pacman -V" workaround:
```
1220 SudoV() {
1221     touch "$tmpdir/sudov.lck"
1222     while [[ -e "$tmpdir/sudov.lck" ]]; do
1223         #sudo -n -v
1224         sudo $pacmanbin -V > /dev/null
1225         sleep 2
1226     done
1227 }
```

**Comment by rmarquis** [comment]
For groups, the following works for me:
```
%wheel ALL=(ALL) NOPASSWD: ALL
```
I did not disable asking password for my user. You might want to check your system settings here. I did not modify anything else regarding permissions.

I'll have a deeper look at that issue shortly, thx for your report.

**Comment by rmarquis** [comment]
Fixed in b811d43e2f9f62f1646c42be2f38a5661c9a29e7.

Tested with:
- `%wheel ALL=(ALL) ALL`
- `%wheel ALL=(ALL) NOPASSWD: ALL`
- `%wheel ALL=(ALL) NOPASSWD: /usr/bin/pacman`
